tests/testthat/test-export-dataset.R

```diff
@@ -69,6 +69,60 @@ with_mock_crunch({
         expect_error(write.csv(ds[c("gender", "birthyr", "gender", "gender", "birthyr")], file=""),
             "Duplicate variable references: gender and birthyr")
     })
+
+    url <- 'https://app.crunch.io/api/datasets/3/export/csv/'
+    post_request <- paste0(
+        '{"filter":null,',
+        '"where":{"function":"select",',
+            '"args":[{"map":{',
+                '"66ae9881e3524f7db84970d556c34552":{"variable":"https://app.crunch.io/api/datasets/3/variables/gender/"},',
+                '"f78ca47313144b57adfb495893968e70":{"variable":"https://app.crunch.io/api/datasets/3/variables/birthyr/"},',
+                '"d7c21314ca9e453c93069168681a285c":{"variable":"https://app.crunch.io/api/datasets/3/variables/starttime/"}}}]'
+)
+    test_that("exporting hidden variables", {
+        ds <- loadDataset("ECON.sav")
+
+        expect_POST(
+            write.csv(ds, file = "", include.hidden = TRUE),
+            url,
+            post_request)
+
+        # ensure that include.hidden is passed down from as.data.frame to write.csv
+        expect_POST(
+            as.data.frame(ds, include.hidden = TRUE, force = TRUE),
+            url,
+            post_request)
+
+        subset_post <- paste0(
+            '{"filter":null,',
+            '"where":{"function":"select",',
+            '"args":[{"map":{',
+                '"66ae9881e3524f7db84970d556c34552":{"variable":"https://app.crunch.io/api/datasets/3/variables/gender/"},',
+                '"f78ca47313144b57adfb495893968e70":{"variable":"https://app.crunch.io/api/datasets/3/variables/birthyr/"}}}]}'
+        )
+        # Hidden variables can be exported by name without include.hidden
+            expect_POST(
+                write.csv(ds[, c("gender", "birthyr")], file = ""),
+                url,
+                subset_post)
+            expect_POST(
+                as.data.frame(ds[, c("gender", "birthyr")], force = TRUE),
+                url,
+                subset_post)
+    })
+    test_that("users can specify all variables by name and get the hidden variables", {
+        skip("TODO modify variablesFilter to distinguish between a fully specified variable subset and the original dataset")
+        # Currently the API makes us send hidden variables as a where clause to
+        # the export API. We work around this by asking for hidden variables when
+        # the user specifies them in the variable subset. We don't currently have a
+        # good way to do this when the user subsets the dataset with all the variable
+        # names. This problem could also go away if the export endpoint allowed us
+        # to pass hidden variable names like regular variable names.
+        expect_POST(
+            write.csv(ds[, c("gender", "birthyr", "starttime")], file = ""),
+            url,
+            post_request)
+    })
```

Comment:
This test should pass already, no? Do we need to skip it?

Reply:
I don't think it does. The reason is that we don't have a way to tell that the user has specified all the column names instead of selecting the whole dataset.